src/coreComponents/common/format/table/unitTests/testTable.cpp

```diff
@@ -289,41 +242,85 @@ TEST( testTable, layoutTable )
              );
 }
 
-TEST( testTable, tableSetMargin )
+TEST( testTable, subColumns )
 {
-  ////////////
-  //////// If setMargin used elsewhere make it public and remove comments for this test
-  ////////////
-  //test with tiny margin
-  // {
-  //   TableLayout tableLayout( {
-  //     TableLayout::ColumnParam{{"Colonne 1"}, TableLayout::Alignment::center},
-  //     TableLayout::ColumnParam{{"Colonne 2"}, TableLayout::Alignment::center},
-  //     TableLayout::ColumnParam{{"Colonne 3"}, TableLayout::Alignment::right},
-  //     TableLayout::ColumnParam{{"Colonne 4"}, TableLayout::Alignment::right},
-  //     TableLayout::ColumnParam{{"Prev\nelement"}, TableLayout::Alignment::right},
-  //     TableLayout::ColumnParam{{"Next\nelement"}, TableLayout::Alignment::right},
-  //   }, "InternalWellGenerator well_injector1" );
-
-  //   //tableLayout.setMargin( TableLayout::MarginValue::tiny );
-
-  //   TableData tableData;
-  //   tableData.addRow( "value 1", "long value 1", "3.0034", 3.0129877, "" , 1 );
-  //   tableData.addRow(  "value 1", "long value 2", "100.45", 4.0129877, 1 , 2 );
-
-  //   TableTextFormatter const tableText( tableLayout );
-  //   EXPECT_EQ( tableText.toString( tableData ),
-// "\n------------------------------------------------------------\n"
-// "|           InternalWellGenerator well_injector1           |\n"
-// "------------------------------------------------------------\n"
-// "|Colonne 1| Colonne 2  |Colonne 3|Colonne 4|   Prev|   Next|\n"
-// "|         |            |         |         |element|element|\n"
-// "------------------------------------------------------------\n"
-// "| value 1 |long value 1|   3.0034|3.0129877|       |      1|\n"
-// "| value 1 |long value 2|   100.45|4.0129877|      1|      2|\n"
-// "------------------------------------------------------------\n\n"
-//              );
-// }
+  {
+    TableLayout const tableLayout( {" ",
+                                    "Column1",
+                                    TableLayout::Column{"Nodes ", TableLayout::Alignment::right, true, {"Locales", "Ghost", "Active"}},
+                                    "Column3",
+                                    TableLayout::Column{"Column4 ", TableLayout::Alignment::right, true, {"Locales", "Ghost"}},
+                                    "Column5"} );
+
+    TableData tableData;
+    tableData.addRow( "min", "125", "375,0001", " YES", 2354654, 562, 43.0, 43.0, 562, 5 );
+    tableData.addRow( "max", "360", "390,1", " YES", 383213213, 712, 48.0, 47.0, 72, 2 );
+
+    TableTextFormatter tableText( tableLayout );
```

Comment:
Would it be hard to refactor to something like this?
```suggestion
    TableLayout tableLayout( {" ",
                              "Column1",
                              TableLayout::Column{"Locales", TableLayout::Alignment::right, true, "Nodes"},
                              TableLayout::Column{"Ghost", TableLayout::Alignment::right, true, "Nodes"},
                              TableLayout::Column{"Active", TableLayout::Alignment::right, true, "Nodes"},
                              "Column3",
                              TableLayout::Column{"Locales", TableLayout::Alignment::right, true, "Column4"},
                              TableLayout::Column{"Ghost", TableLayout::Alignment::right, true, "Column4"},
                              "Column5"} );
```
We would have a "parent column" last parameter, it would not be closed to further sub-sub-column management (through something like "level1 Parent Column/level2 Parent Column")

Reply:
If needed i could go for this but rn it will take some time to impl this